app/assets/javascripts/discourse/app/controllers/preferences/account.js

```diff
@@ -147,6 +164,14 @@ export default Controller.extend(CanCheckEmails, {
         .catch(popupAjaxError);
     },
 
+    setPrimaryEmail(email) {
+      this.model.setPrimaryEmail(email).then(() => {
+        this.model.secondary_emails.removeObject(email);
```

Comment:
shouldn't we do all of this this from inside `setPrimaryEmail` in models/user.js file given it's all model logic ?